File: PCube/src/app/models/timelineFilter.ts
export class TimelineFilterItem{
    day_of_week : string;
    project_name: string;
    accounting_time_category_name: string;
    activity_name: string;
    member_name: string;

    constructor() {
        this.day_of_week = "";
        this.project_name = "";
        this.accounting_time_category_name = "";
        this.activity_name = "";
        this.member_name = "";
    }
}

File: PCube-API/domain/timelineFilter.py
class TimelineFilter:

    def __init__(self):
        self.day_of_week = ""
        self.project_name = ""
        self.accounting_time_category_name = ""
        self.activity_name = ""
        self.member_name = ""

    def asDictionary(self):
        return {
            "day_of_week": self.day_of_week,
            "project_name": self.project_name,
            "accounting_time_category_name":
                self.accounting_time_category_name,
            "activity_name": self.activity_name,
            "member_name": self.member_name
        }
